### PR TITLE
11 testimonial block

### LINKED ISSUE #11: Testimonial
New component for homepage and other pages

Variants:
- [with video](https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?type=design&node-id=2244-171462&mode=design&t=Ui58kA2zbVJpDxoC-4) vs. [only image](https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?type=design&node-id=2243-159054&mode=design&t=Ui58kA2zbVJpDxoC-4)
- media right vs. [media left](https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?type=design&node-id=2232-60938&mode=design&t=Ui58kA2zbVJpDxoC-4), on mobile the image will be shown first
- overlap vs. no overlap

With video: footer will be added to contain author. When there’s too much content, the [author text will overflow](https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?type=design&node-id=2243-99518&mode=design&t=Ui58kA2zbVJpDxoC-4). Play button added in this footer. 

The play button will open a modal window with the video.

Image

### PR DESCRIPTION
**Pull request by TomaszDziezykNetcentric**
Fix #11 

Test URLs:
- Before: https://main--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/tdziezyk/v2/testimonial
- After: https://11-testimonial-block--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/tdziezyk/v2/testimonial

### CODE REVIEW

File: blocks/v2-testimonial/v2-testimonial.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  const columns = block.querySelectorAll(':scope > div > div');
+  block.parentElement.classList.add('full-width');
+
+  columns.forEach((col) => {
+    col.classList.add(`${blockClass}__column`);
+
+    const headings = [...col.querySelectorAll('h1, h2, h3, h4, h5, h6')];
+    headings.forEach((h) => h.classList.add(`${blockClass}__heading`));
+
+    const firstHeading = headings[0];
```
Comment (kesiah) [inline blocks/v2-testimonial/v2-testimonial.js:56]: We can simplify this to `headings[0]?.classList.add('with-marker');`

File: blocks/v2-testimonial/v2-testimonial.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  if (!videosLinks.length) {
+    return;
+  }
+
+  handleVideoLinks(videosLinks, block);
+  col.classList.add(`${blockClass}__video-section`);
+  col.setAttribute('data-theme', 'gold');
+  col.querySelector('p').classList.add(`${blockClass}__author`);
+  const videoLinkEl = col.querySelector('.v2-testimonial__video-link');
```
Comment (kesiah) [inline blocks/v2-testimonial/v2-testimonial.js:39]: `
const videoLinkEl = col.querySelector(`.${blockClass}__video-link`);
`

File: blocks/v2-testimonial/v2-testimonial.css [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+.v2-testimonial__video-link .icon {
+  height: 24px;
+}
+
+.v2-testimonial__video-link svg {
+  height: 24px;
+  width: 24px;
+}
+
+.v2-testimonial__text-row {
```
Comment (kesiah) [inline blocks/v2-testimonial/v2-testimonial.css:113]: Is the order necessary? If we are positioning the elements absolutely.

**Review by kesiah — CHANGES_REQUESTED** [review]

File: blocks/v2-testimonial/v2-testimonial.css (hunk 3 above)
Comment (TomaszDziezykNetcentric) [reply blocks/v2-testimonial/v2-testimonial.css:113]: On mobile, there is no absolute positioning. The order should always be testimonial text at the top and image at the bottom - it doesn't depend on the doc's content.

File: blocks/v2-testimonial/v2-testimonial.js (hunk 1 above)
Comment (TomaszDziezykNetcentric) [reply blocks/v2-testimonial/v2-testimonial.js]: Updated

File: blocks/v2-testimonial/v2-testimonial.js (hunk 2 above)
Comment (TomaszDziezykNetcentric) [reply blocks/v2-testimonial/v2-testimonial.js]: Updated

**Review by cogniSyb — APPROVED** [review]

**Review by Lakshmishri — CHANGES_REQUESTED** [review]

File: blocks/v2-testimonial/v2-testimonial.css [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  padding-right: calc(100vw - 510px);
+}
+
+.v2-testimonial__blockquote-column strong {
+  font-family: var(--ff-body-bold);
+}
+
+.v2-testimonial__heading {
+  color: var(--c-primary-white);
+  font-family: var(--ff-headline-medium);
```
Comment (Lakshmishri) [inline blocks/v2-testimonial/v2-testimonial.css:28]: Let's use font variables defined here with font shorthand. Can remove font-style as it has no effect
` font: var(--headline-1-font-size) var(--ff-headline-medium);
   line-height: var(--headline-1-line-height);`
   
   Using short hand is optional

File: blocks/v2-testimonial/v2-testimonial.css [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+  font-size: 32px;
+  font-style: normal;
+  line-height: 115%;
+  letter-spacing: -0.64px;
+  margin: 24px 0;
+}
+
+.v2-testimonial__blockquote {
+  font-family: var(--ff-testimonial);
```
Comment (Lakshmishri) [inline blocks/v2-testimonial/v2-testimonial.css:37]: Same as above use font varaibles

File: blocks/v2-testimonial/v2-testimonial.css [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+  flex-direction: column;
+}
+
+.v2-testimonial__blockquote-column {
+  background-color: #2b2b2b;
+  padding: 48px 16px;
+  color: var(--c-primary-white);
+  font-family: var(--ff-body);
+  font-size: 14px;
+  font-style: normal;
```
Comment (Lakshmishri) [inline blocks/v2-testimonial/v2-testimonial.css:15]: You can remove font-style and font-weight

**Review by Lakshmishri — CHANGES_REQUESTED** [review]

File: common/modal/modal-component.js [hunk 7]
```diff
@@ -36,7 +36,17 @@ const createModal = () => {
 
   // adding close modal button
   const closeButton = createElement('button', ['modal-close-button']);
-  const closeIcon = createElement('i', ['fa', 'fa-close']);
+  const closeIcon = createElement('span', ['icon', 'icon-close']);
+  closeIcon.innerHTML = `
+  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (TomaszDziezykNetcentric) [reply common/modal/modal-component.js]: Done

File: blocks/v2-testimonial/v2-testimonial.css (hunk 4 above)
Comment (TomaszDziezykNetcentric) [reply blocks/v2-testimonial/v2-testimonial.css]: Updated

File: blocks/v2-testimonial/v2-testimonial.css (hunk 5 above)
Comment (TomaszDziezykNetcentric) [reply blocks/v2-testimonial/v2-testimonial.css]: Updated

File: blocks/v2-testimonial/v2-testimonial.css (hunk 6 above)
Comment (TomaszDziezykNetcentric) [reply blocks/v2-testimonial/v2-testimonial.css]: Updated

**Review by Lakshmishri — APPROVED** [review]

File: common/modal/modal-component.js [hunk 8]
```diff
@@ -36,7 +36,18 @@ const createModal = () => {
 
   // adding close modal button
   const closeButton = createElement('button', ['modal-close-button']);
-  const closeIcon = createElement('i', ['fa', 'fa-close']);
+  const closeIcon = createElement('span', ['icon', 'icon-close']);
+  const svgCloseIcon = document.createRange().createContextualFragment(`
+    <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+      <g id="Icons/16px/Plus">
```
Comment (kesiah) [inline common/modal/modal-component.js:42]: I'm pretty sure the groups can be removed. Removing Id's is really important, we might end up with repeated ID's in HTML :)

**Review by kesiah — CHANGES_REQUESTED** [review]

File: common/modal/modal-component.js (hunk 8 above)
Comment (TomaszDziezykNetcentric) [reply common/modal/modal-component.js]: Yup, you're right. Changed.

**Pull request merged**